product_variant_available_in_pos/models/product_template.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, models
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = 'product.template'
+
+    def _update_available_in_pos(self, vals):
+        values = {}
+        if 'available_in_pos' in vals:
+            values['available_in_pos'] = vals['available_in_pos']
+        if values:
+            self.product_variant_ids.write(values)
+
+    @api.multi
+    def write(self, vals):
+        res = super(ProductTemplate, self).write(vals)
+        for product in self:
```

Comment:
I believe it would be better to move the if statement actually present in the method '_update_available_in_pos' into the 'write' method just above the for statement.
`    def _update_available_in_pos(self, vals):
        values = {'available_in_pos': vals['available_in_pos']}
        self.product_variant_ids.write(values)`

    `@api.multi
    def write(self, vals):
        res = super(ProductTemplate, self).write(vals)
        if 'available_in_pos' in vals:
            for product in self:
                product._update_available_in_pos(vals)
        return res`

This very small change wouldn't change the functionality of the involved methods but could slightly improve the performance, computing the expression in the if statement once for all instead than as many times as the records in self (notice that its value won't be changed during the iteration).